system/security/securesocket/securesocket.cpp

```diff
@@ -1961,51 +1992,61 @@ extern "C" {
 }
 
-SECURESOCKET_API ISecureSocketContext* createSecureSocketContextEx2(const IPropertyTree* config, SecureSocketType sockettype)
+
+SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSecret(const char *issuer, SecureSocketType sockettype)
 {
-    if (config == NULL)
-        return createSecureSocketContext(sockettype);
+    Owned<const ISyncedPropertyTree> info = getIssuerTlsSyncedConfig(issuer);
+    return createSecureSocketContextSynced(info, sockettype);
+}
 
-    return new securesocket::CSecureSocketContext(config, sockettype);
-}       
 
-SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSSF(ISmartSocketFactory* ssf)
+SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSecretSrv(const char *issuer, bool requireMtlsFlag)
 {
-    if (ssf == nullptr || !ssf->queryTlsConfig())
-        return createSecureSocketContext(ClientSocket);
+    if (requireMtlsFlag && !queryMtls())
+        throw makeStringException(-100, "TLS secure communication requested but not configured");
+
+    Owned<const ISyncedPropertyTree> info = getIssuerTlsSyncedConfig(issuer);
+    if (!!info->isValid())
+        throw makeStringException(-101, "TLS secure communication requested but not configured (2)");
 
-    return new securesocket::CSecureSocketContext(ssf->queryTlsConfig(), ClientSocket);
+    return createSecureSocketContextSynced(info, ServerSocket);
 }
 
-SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSecret(const char *issuer, SecureSocketType sockettype)
+IPropertyTree * createSecureSocketConfig(const char* certFileOrBuf, const char* privKeyFileOrBuf, const char* passphrase)
 {
-    Owned<IPropertyTree> info = getIssuerTlsServerConfig(issuer);
-    //if the secret doesn't exist doesn't exist just go on without it. IF it is required the tls connection will fail. 
-    //This is primarily for client side... server side would probably use the explict ptree config or explict cert param at least for now.
-    if (info)
-        return createSecureSocketContextEx2(info, sockettype);
-    else
-        return createSecureSocketContext(sockettype);
+    if (!certFileOrBuf && !privKeyFileOrBuf && !passphrase)
+        return nullptr;
+
+    Owned<IPropertyTree> config = createPTree("ssl");
+    if (certFileOrBuf)
+        config->setProp("certificate", certFileOrBuf);
+    if (privKeyFileOrBuf)
+        config->setProp("privatekey", privKeyFileOrBuf);
+    if (passphrase)
+        config->setProp("passphrase", passphrase);
+    return config.getClear();
 }
 
-SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSecretSrv(const char *issuer, bool requireMtlsFlag)
+SECURESOCKET_API ISecureSocketContext* createSecureSocketContextSSF(ISmartSocketFactory* ssf)
 {
-    if (requireMtlsFlag && !queryMtls())
-        throw makeStringException(-100, "TLS secure communication requested but not configured");
+    if (ssf == nullptr || !ssf->queryTlsConfig())
```

Comment:
could remove the second condition, since it will fall through the the same code below.